### ISSUE TITLE
Fix delete / cleanup of kind cluster when standalone-cluster or management-cluster delete fails

### ISSUE THREAD

**Issue by karuppiah7890** [vmware-tanzu/community-edition#2533]
Fix delete / cleanup of kind cluster when standalone-cluster or management-cluster delete fails. Currently this is not being done, for example -

https://github.com/vmware-tanzu/community-edition/blob/0bf658cbbb3ce8b5469f2ff76a04bbdb8ce2aa4a/test/azure/deploy-standalone-cluster.sh#L63-L70

https://github.com/vmware-tanzu/community-edition/blob/0bf658cbbb3ce8b5469f2ff76a04bbdb8ce2aa4a/test/azure/deploy-management-and-workload-cluster.sh#L278-L281

Also, as part of this issue, we can also fix unnecessary kind cluster delete processes. For example I made a mistake adding this extra `delete_kind_cluster` here -

https://github.com/vmware-tanzu/community-edition/blob/0bf658cbbb3ce8b5469f2ff76a04bbdb8ce2aa4a/test/vsphere/run-tce-vsphere-standalone-cluster.sh#L63-L68

Ideally, if `tanzu standalone-cluster create` is working correctly, it would have deleted the kind bootstrap cluster once the standalone cluster creation is done. So, when it comes to `check-tce-cluster-creation.sh`, the kind bootstrap cluster would ideally not be present, but I added the `delete_kind_cluster` due to some confusion with #2457 issue and it's PR #2459 . So the `delete_kind_cluster` after `check-tce-cluster-creation.sh` but it doesn't cause any problems in a clean CI environment, no failures etc, as we use `xargs` -

https://github.com/vmware-tanzu/community-edition/blob/0bf658cbbb3ce8b5469f2ff76a04bbdb8ce2aa4a/test/util/utils.sh#L10-L13

But yeah, in local (dev machine) it would unnecessarily delete some kind bootstrap clusters, which is also a generic problem mentioned here - #1270 with a possible solution but it hasn't been implemented yet, so I guess that's a separate issue but I just wanted to call it out here. But in any case, the `delete_kind_cluster` just right after `check-tce-cluster-creation.sh` failure is not necessary

**Comment by karuppiah7890** [comment]
Till now we haven't seen any problems due to the issue, so there's no evidence to show that the kind cluster cleanup didn't happen, say in CI. Maybe in local this can be simulated. But it's important to cleanup things to not bloat up the environment - be it in CI (ephemeral or non-ephemeral) or dev.

**Comment by karuppiah7890** [comment]
I missed to simulate the scenario and test it. Looks like there are still some problems in the PR #2548 that I created, causing the cleanup to not happen while other failures happen, for example, pod ready timeout and not during deletion failure

**Comment by karuppiah7890** [comment]
Some pipelines that got affected and failed -

https://github.com/vmware-tanzu/community-edition/runs/4289831772?check_suite_focus=true - `test-mc-9355`

https://github.com/vmware-tanzu/community-edition/runs/4289904215?check_suite_focus=true - `test-mc-18356`

**Comment by karuppiah7890** [comment]
Due to this issue, there were many Azure TCE clusters lying around that weren't cleaned up 😅 . Cluster unique names that can be found in the pipelines -

```
test-mc-5871
test-mc-4786
test-mc-201
test-mc-25169
test-wld-25169
test-mc-13798
test-wld-13798
test-mc-23235
test-wld-23235
test-mc-4801
test-mc-4604
```
